packages/flutter/test/material/tabs_test.dart

```diff
@@ -600,6 +600,51 @@ void main() {
     expect(find.text('RIGHT CHILD'), findsNothing);
   });
 
+  // A regression test for https://github.com/flutter/flutter/pull/88878.
+  testWidgets('TabController notifies the index to change when left flinging', (WidgetTester tester) async {
+    final List<String> tabs = <String>['A', 'B', 'C'];
+    late TabController tabController;
+
+    Widget buildTabControllerFrame(BuildContext context, TabController controller) {
+      tabController = controller;
+      return MaterialApp(
+        theme: ThemeData(platform: TargetPlatform.iOS),
+        home: Scaffold(
+          appBar: AppBar(
+            title: const Text('tabs'),
+            bottom: TabBar(
+              controller: controller,
+              tabs: tabs.map<Widget>((String tab) => Tab(text: tab)).toList(),
+            ),
+          ),
+          body: TabBarView(
+            controller: controller,
+            children: const <Widget>[
+              Center(child: Text('CHILD A')),
+              Center(child: Text('CHILD B')),
+              Center(child: Text('CHILD C')),
+            ],
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(TabControllerFrame(
+      builder: buildTabControllerFrame,
+      length: tabs.length,
+      initialIndex: tabs.indexOf('C'),
+    ));
+    expect(tabController.index, tabs.indexOf('C'));
+
+    tabController.addListener(() {
+      final int indexOfB = tabs.indexOf('B');
+      expect(tabController.index == indexOfB, true);
+    });
+    final Offset flingStart = tester.getCenter(find.text('CHILD C'));
+    await tester.flingFrom(flingStart, const Offset(600, 0.0), 10000.0);
```

Comment:
Something is not right either with this test or my understanding.  I would have thought that the `600` here could have been less than 200, the value used in the [test that I gave as an example](https://github.com/flutter/flutter/blob/1d862ec7eaaa3e34a162c53b5d504c11d10543fa/packages/flutter/test/material/tabs_test.dart#L519), because this PR allows the detection of smaller page flings.  Is that right?

Also, if I put a log in your tabController listener, I see that it is called twice.  I'm not sure why that is, maybe because the fling is so big?

Reply:
@justinmc 
Sorry late response 🙏 
 
 > I see that it is called twice.
 
I think this behavior is as normal. When I made the video for this PR, the listener was called twice for a fling.
 
> this PR allows the detection of smaller page flings. Is that right?
 
I couldn't reproduce the pattern with small fling. 
When We do a big swipe, the page value sometimes overflows from the target index, and I think that's the cause of this problem.
 
page overflows with big fling |
:-: | 
<img src='https://user-images.githubusercontent.com/19836917/134016428-1b3c49d6-ced7-433f-9796-65589b4c53e9.jpg' width='400px'> |

Reply:
Got it, thank you for the thorough response and the diagram!  I was missing the fact that this fling was more than one whole item in length.  Makes sense.  I also confirmed that this test fails on master (it gives `0` on master, as expected from your diagram).